src/Core/Grid/Data/Factory/DoctrineGridDataFactory.php

```diff
@@ -62,6 +80,12 @@ public function getData(SearchCriteriaInterface $searchCriteria)
         $records = $searchQueryBuilder->execute()->fetchAll();
         $recordsTotal = (int) $countQueryBuilder->execute()->fetch(PDO::FETCH_COLUMN);
 
+        $this->hookDispatcher->dispatchWithParameters('action'. $this->gridId .'GridQueryBuilderModifier', [
+            'grid_id' => $this->gridId,
```

Comment:
i guess we dont need this param, right? or if we keep it, we can remove grid id from hook name, that would solve issue with definition id having `_`?

Reply:
? we need the definition Id in the hook, or your hook will impact ALL grids at the same time

Reply:
not really, you would have to check for grid id in your hook, similar approach is done it some other hooks.

Reply:
I don't want that, I want a specific hook for each grid. If you need to filter the conditions of using a hook, it's already a defeat regarding the developer experience.

Reply:
please stick to the documented schema, and tell me if something is not possible :)

Reply:
ok, then lets remove `grid_id` param. :+1: